sale_order_warehouse_location/models/sale_order.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2025 Manuel Regidor <manuel.regidor@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.depends("partner_shipping_id")
+    def _compute_warehouse_id(self):
+        ret_vals = super()._compute_warehouse_id()
+        for order in self.filtered(
+            lambda o: o.partner_shipping_id
+            and (o.state in ["draft", "sent"] or not o.ids)
+        ):
+            warehouses = self.env["stock.warehouse"].search(
```

Comment:
I don't understand this. Why don't you create a function in warehouse that returns the available warehouses directly?

Reply:
In that case I'd have to pass an additional parameter to the warehouses_by_country_state function in stock.warehouse, which I think is unnecessary if the warehouses related to the company are previously selected.